### ISSUE TITLE
Create repo in stack A and ref it to codepipeline in stack B throws 'cyclic reference'

### ISSUE THREAD

**Issue by codyfy** [aws/aws-cdk#5727]
### The Question

Stack A has a CodeCommit repo. I want the CodePipeline in Stack B to use this repo. This throws `Adding this dependency (cdk-repo -> cdk-pipe/CodePipeline/Resource.Ref) would create a cyclic reference.`

I just had a very similar problem #5657 where I manually needed to hard code the artifact bucket. However, this time the resource mentioned in the error is hardcoded. So I am unsure if this me not understanding how this is supposed to work or if this is unexpected behavior.

### Code 
[Here](https://github.com/codyfy/aws-repo-pipeline-cyclic-reference) is a repo with an example of what I try to do.

app.py
```
#!/usr/bin/env python3

from aws_cdk import core

from cdk_repopipe.cdk_repo_stack import RepoStack
from cdk_repopipe.cdk_pipe_stack import PipelineStack


app = core.App()
repo_stack = RepoStack(app, "cdk-repo")
pipe_stack = PipelineStack(app, "cdk-pipe", repo_stack.repo)

app.synth()
```
repo.py
```
from aws_cdk import (
    core,
    aws_codecommit as codecommit
)


class RepoStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # The code that defines your stack goes here
        self.repo = codecommit.Repository(self, "my-repo", repository_name="test-repo")
```
pipeline.py
```
from aws_cdk import (
    core,
    aws_codecommit as codecommit,
    aws_codebuild as codebuild,
    aws_codepipeline as codepipeline,
    aws_autoscaling as autoscaling,
    aws_codepipeline_actions as codepipeline_actions,
    aws_codedeploy as codedeploy,
    aws_ec2 as ec2,
    aws_s3 as s3
)


class PipelineStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, repo: codecommit.Repository, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Pipeline
        artifactBucket = s3.Bucket(self, 'PipelineBucket',
            bucket_name="bucket-name")

        pipeline = codepipeline.Pipeline(self, "CodePipeline",
            artifact_bucket=s3.Bucket.from_bucket_attributes(self, 'ImportedBucket',
            bucket_name="bucket-name")
        )
        
        # Source
        source_output = codepipeline.Artifact()
        source_action = codepipeline_actions.CodeCommitSourceAction(
            action_name="Source",
            repository=repo,
            output=source_output
        )        
        pipeline.add_stage(stage_name="Source", actions=[source_action])


        # Deploy
        deploy_application = codedeploy.ServerApplication(self, "CodeDeployApplication", 
            application_name="application-name"
        )

        deployment_group = codedeploy.ServerDeploymentGroup(self, "DeploymentGroup",
            application=deploy_application,
        )
        
        deploy_action = codepipeline_actions.CodeDeployServerDeployAction(
            action_name="deploy",
            input=source_output,
            deployment_group=deployment_group
        )
        pipeline.add_stage(stage_name="Deploy", actions=[ deploy_action ])
```

### Environment

  - **CDK CLI Version:** aws-cli/1.14.44 Python/3.6.9 Linux/5.0.0-37-generic botocore/1.13.13
  - **Module Version:** 1.19.0 (build 5597bbe)
  - **OS:** Ubuntu 18.04.3 LTS
  - **Language:** Python

### Other information

The repo has to be in a different stack so it can be used by prod and dev stacks

**Comment by markusl** [comment]
I'm also encountering the same issue. Is there any workaround for this behavior until a proper fix is implemented?

**Comment by skinny85** [comment]
Hey @codyfy  and @markusl ,

yes - there should be a simple answer to this! The cycle is because of a CloudWatch Event that is added to the repository that triggers the pipeline. If you change the source action to use polling instead:

```python
        source_action = codepipeline_actions.CodeCommitSourceAction(
            # rest of the properties as above...
            trigger=codepipeline_actions.CodeCommitTrigger.POLL,
        )
```

this will remove the cycle.

Let me know if this works!

Thanks,
Adam

**Comment by mikestopcontinues** [comment]
Also, you can reference the repo by name, and add a manual dependency from the Pipeline stack to the stack with the repo:

```js
// App
const repositoryName = 'Repo';
const app = new App();

// CommonStack
const commonStack = new Stack(app, 'CommonStack', stackProps);
commonStack.repo = new Repository(commonStack, 'CommonCodeRepo', {repositoryName});

// PipelineStack
const pipeStack = new Stack(app, 'PipelineStack', stackProps);
pipeStack.addDependency(commonStack);
pipeStack.repo = Repository.fromRepositoryName(pipeStack, 'SourceRepo', repositoryName);
```

**Comment by skinny85** [comment]
Yes, this is excellent advice @mikestopcontinues!

**Comment by ccurrie-amzn** [comment]
I'd like to request that this be re-opened. If I take the OP's code and replace the direct use of the repository with `codecommit.Repository.fromRepositoryArn(self, "ImportedRepository", repository.repositoryArn)`, the reference and the cycle are not created. Is there a barrier to the pipeline or the source action detecting that the repository is from a different stack, and "doing the right thing" in this case?

**Comment by skinny85** [comment]
Re-opening as per @ccurrie-amzn 's request.

**Comment by teddyaryono** [comment]
Is there any update to this one?

**Comment by skinny85** [comment]
Not much @teddyaryono , sorry. A PR would be very welcome if that's a possibility.

**Comment by athempel** [comment]
Is this a duplicate of #3087?

**Comment by skinny85** [comment]
Yes, I believe this issue was fixed by PR #20149, the same PR that fixed #3087.
